mender-update/standalone/standalone.cpp

```diff
@@ -495,6 +581,16 @@ ResultAndError DoCommit(
 		return_err = return_err.FollowedBy(err);
 	}
 
+	// Commit Leave
+	{
+		auto err =
+			script_runner.RunScripts(executor::State::ArtifactCommit, executor::Action::Leave);
+		if (err != error::NoError) {
+			log::Error("Error during Commit Leave script (ignored): " + err.String());
+		}
```

Comment:
I used this as my guide:

```
For most of the states, if some error occurs while executing either an Enter or a Leave script, or some action inside the state (like installing a new artifact which is broken and therefore installation is failing) the corresponding Error scripts are executed. The exceptions are:

    Idle
    Sync
    ArtifactCommit_Leave
    ArtifactRollback
    ArtifactRollbackReboot
    ArtifactFailure
    
    The reason for ignoring most of these and not calling Error scripts is that the state is already an error state, such as for example ArtifactRollback. ArtifactCommit_Leave is not an error state, but in this case it is too late to go to an Error script, because the Artifact has already been committed.
```

I may have misread it, but it's pretty easy to think that this means that `Error` scripts are not executed in this instance I think.

Reply:
Here's what I found in practice while testing the Golang client:

* **Idle**: `Enter` and `Leave` errors are ignored, `Error` scripts are never called.
* **Sync**: `Enter` and `Leave` errors are respected, `Error` scripts are called.
* **ArtifactCommit_Leave**: `Enter` and `Leave` errors are respected in the deployment status, however `Error` scripts are *not* called for errors following `Leave`. IMHO, this is a bug, it should call them.
* The other work in the expected way, errors are ignored and `Error` scripts are not called.

Does that make sense? And should I update the docs?

Reply:
Certainly not against it, tbh I would expect errors to run in all states.

I am a little curious though, as it goes against the logic written out in the docs, where `The state is already in an error state`. 

But I always found this a little odd, as it breaks the symmetry of our implementation, whereas all states do have `Error` support, which I kinda like.

But for now, I will go ahead with what you listed.

Reply:
> I am a little curious though, as it goes against the logic written out in the docs, where `The state is already in an error state`.

Yes, but I will change this, since it's misleading. It should apply only to the states that truly are on the error path (which does not include `Idle`, `Sync`, and `ArtifactCommit_Leave`.